File: frontend/src/data/customers/Card.ts
export default interface Card {
  cardId: string;
  debit: number;
  active: boolean;
}

export interface SummaryCard {
  cardId: number;
  debit: number;
}

export interface RequestCard {
  cardId: string;
}

export const CardStatusMetadata: Record<number, { pt: string; en: string }> = {
  1: { pt: "Ativo", en: "Active" },
  0: { pt: "Desativado", en: "Deactivated" },
};

File: backend/src/main/java/com/storecontrol/backend/models/customers/Card.java
package com.storecontrol.backend.models.customers;

import com.storecontrol.backend.models.customers.request.RequestCard;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@Entity
@Table(name = "cards")
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Card {

    @Id @Column(name = "card_id")
    private String id;

    @Column(nullable = false)
    private BigDecimal debit;

    @OneToMany(mappedBy = "card")
    private List<Customer> customers;

    @Column(nullable = false)
    private boolean active;


    public Card(RequestCard request) {
        this.id = request.cardId();
        this.debit = BigDecimal.ZERO;
        this.active = false;
    }

    public void updateActive(Boolean value) {
        this.active = value;
    }

    public void incrementDebit(BigDecimal value) {
        this.debit = debit.add(value);
    }
}
